pre-commit

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
+# original source: https://raw.githubusercontent.com/streadway/amqp/master/pre-commit
+
+GOFMT_OK=1
+GOLINT_OK=1
+GOVET_OK=1
+
+main() {
+    mv_vendor_out
+    run_gofmt
+    run_golint
+    mv_vendor_in
+    run_govet
+    check_results
+}
+
+local_go_version_is_latest_stable() {
```

Comment:
I could drop this function since I'm not using it any more. The original source has a wrapper in the `main()` function to exit if the local Golang version doesn't match the constant set to a particular Golang version. In this case, I didn't care that much as I just want to see the thing passing. Not sure we have a specific version of Golang we're tying ourselves to currently.